From ordex@autistici.org Fri Nov 16 16:49:16 2012
X-Apparently-To: lindner_marek@yahoo.de via 188.125.84.191; Fri, 16 Nov 2012 08:50:21 +0000
Received-SPF: pass (domain of autistici.org designates 178.255.144.35 as permitted sender)
X-YMailISG: fVfMssIWLDsEjVl9FTNnqLYIz03Fp5zMnBfNrwq01TpmFHMe
 4NioQil50ZX.KNrDwxWrNL7F5LVE3iFuNBiOj4vXGfU9GDPuCH8xLBGseyad
 EatFIUDFeUMbfEwZPy28z_9EXlsHz5RAaq3WJlVwxlbl8HYO8KDoP.O2uPJB
 f1LCApHhOzbwEorY6aey0roQgYwXheDamyypNW23KoTrKRRt2yIX_44YOEX6
 yPbmxmK2m2L048nZkBe5u1LD89PXTn2P07.Km_O95JLxjx1PCFbKa3TBNIk5
 n367Buu6KbhXuIU39l_ilvI6UWIwyDT9gyhYrjDgPn8680zLGVtcXjUO9DLw
 lHybXd6kneRWA7krXtrHIxRPJrayaHbsMEztJWBGUMrwglJlJThMRvp8nhgV
 nw0fVqbQI7sUiFMZx9QVMETsQ2krXG7GUVNFuELV_ej3IP_vjYBZtLBB8Aqb
 Wh9yk0UByNlHv82yvc_N3B67fzOEcJkC6fsFuvC7oBsjFBU.j6ejX_BrPiyi
 M7uelmPRLdOujZpZ.7K3Az1nCgdyOcWuf.F37.409a2DQJDh8EuptbFF0gJR
 p4nscDuQUch1VOaYVXUzgIFpmVNQbbvV63t2PegzhrYV1xmv8TXv4DPDDmKw
 7zEqGuOok9HayiISSYrlxLiw4a2OQoaNhGPtoFzxVdWJxReku9T6yB8tedcW
 ilpf3_Rf1MDCB2W1HxRspPq6RcSu2WleYAINgzrlU6Gsgks2zExjTuYVaIv0
 9ydk8.K77fsbS11Nmj6cynKgooOZiHpao2I6ggYO5tTM4_R4slHPVZlg8hGB
 feKVfKaeYKPYQ3chD99PuxjXoEup9Cn3rmRcWNDMd72ncd1Xk43HHkDyTmSI
 6OXRkTWMEkEhqA6MT7CVOuWnzRaT6AxKBybOso7XWRaiIxi1AOpEy7ihWLNW
 pcQkRakzv2wqpknWLMlgHzLZs3LEPSpcCQwHBbVMkLL6S8m_moDLO4fdzz0v
 i40nJUsBpxOAl49u3ZTM0G2ccDz5IJVOTuyqbGtZdeG5MBen9DJh4NCYP_No
 psAEk9j1XV_1aaAakSNFXnS18XtYBtZaewE.O5KTHddFyFSBWF2VwR2dFEIf
 NV9kSDSl6Gm4P2PjYWOdeXf0YiRm6cba_Q--
X-Originating-IP: [178.255.144.35]
Authentication-Results: mta1027.mail.ukl.yahoo.com  from=autistici.org; domainkeys=neutral (no sig);  from=autistici.org; dkim=pass (ok)
Received: from 127.0.0.1  (EHLO contumacia.investici.org) (178.255.144.35)
  by mta1027.mail.ukl.yahoo.com with SMTP; Fri, 16 Nov 2012 08:50:21 +0000
Received: from [178.255.144.35] (contumacia [178.255.144.35]) (Authenticated sender: ordex@autistici.org) by localhost (Postfix) with ESMTPSA id CEF29E8797;
	Fri, 16 Nov 2012 08:50:19 +0000 (UTC)
X-DKIM: Sendmail DKIM Filter v2.8.2 contumacia.investici.org CEF29E8797
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed; d=autistici.org;
	s=stigmate; t=1353055821;
	bh=Z+GuGLsx9ZvEgtokRJqUrcaZ46uGZCSjpzl9VagyFsY=;
	h=From:To:Cc:Subject:Date:Message-Id:In-Reply-To:References;
	b=daWWrrrXRKMymNqgCbTU4Z67tlYH+UtOkzrTMpikDtsSVqGhdWuKPG/mY2+eaO+ra
	 ap5dS/KrwlFmxni3y16IuXKcPQo8EIANkE0mWSXZgOOqh9ps3VrYZBlWoThf8w/NYw
	 Mw3Bi7uuS75kwEq5akDKroDq8jufp9RVZ+ZVW3oY=
From: Antonio Quartulli <ordex@autistici.org>
To: davem@davemloft.net
Cc: netdev@vger.kernel.org,
 Simon Wunderlich <simon.wunderlich@s2003.tu-chemnitz.de>,
 Marek Lindner <lindner_marek@yahoo.de>,
 Sven Eckelmann <sven@narfation.org>,
 Antonio Quartulli <ordex@autistici.org>
Subject: [PATCH] batman-adv: correctly pass the client flag on tt_response
Date: Fri, 16 Nov 2012 09:49:16 +0100
Message-Id: <1353055758-2901-3-git-send-email-ordex@autistici.org>
X-Mailer: git-send-email 1.8.0
In-Reply-To: <1353055758-2901-1-git-send-email-ordex@autistici.org>
References: <1353055758-2901-1-git-send-email-ordex@autistici.org>
Status: R
X-Status: N
X-KMail-EncryptionState:  
X-KMail-SignatureState:  
X-KMail-MDN-Sent:  

When a TT response with the full table is sent, the client flags
should be sent as well. This patch fix the flags assignment when
populating the tt_response to send back

This was introduced by 30cfd02b60e1cb16f5effb0a01f826c5bb7e4c59
("batman-adv: detect not yet announced clients")

Signed-off-by: Antonio Quartulli <ordex@autistici.org>
Signed-off-by: Marek Lindner <lindner_marek@yahoo.de>
---
 translation-table.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/translation-table.c b/translation-table.c
index 64c0012..fec1a00 100644
--- a/translation-table.c
+++ b/translation-table.c
@@ -1502,7 +1502,7 @@ batadv_tt_response_fill_table(uint16_t tt_len, uint8_t ttvn,
 
 			memcpy(tt_change->addr, tt_common_entry->addr,
 			       ETH_ALEN);
-			tt_change->flags = BATADV_NO_FLAGS;
+			tt_change->flags = tt_common_entry->flags;
 
 			tt_count++;
 			tt_change++;
-- 
1.8.0

